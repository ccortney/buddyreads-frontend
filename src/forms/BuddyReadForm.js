const BuddyReadForm = () => {
    return (
        <h2>Buddy Read Form</h2>
    )
}

export default BuddyReadForm;