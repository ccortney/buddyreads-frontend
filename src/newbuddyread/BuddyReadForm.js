import { Grid, Box, TextField, Button } from "@mui/material"

/** New Buddy Read Form
 *
 * Displays form to create new buddy read and send an invite to buddy.
 * Handles changes to local form state.
 * Submitting the form calls the API to create new buddy read. 
 *
 * Confirmation of a successful invite sent is normally a simple <ErrorAlert>
 *
 * Routed as /buddyreads/new
 * Result -> NewBuddyReadForm -> ErrorAlert
 */

const BuddyReadForm = ({formData, handleFormChange, handleFormSubmit}) => {

    return (
        <Grid container align='center' sx={{justifyContent:'center'}}>
            <form onSubmit={handleFormSubmit}>
                <Box sx={{ display: 'flex', justifyContent:  'center' }}>
                    <TextField
                        label="Buddy's Email"
                        id="buddy"
                        name="buddy"
                        value={formData.buddy}
                        onChange = {handleFormChange}
                        sx={{ m: 1, width: '25ch' }}
                    />
                    <Box sx={{display: 'flex', alignItems: 'center',  m: 1}}>
                        <Button variant="contained" type="submit">Invite</Button>
                    </Box>
                </Box>
            </form>
        </Grid>
    )
}

export default BuddyReadForm;