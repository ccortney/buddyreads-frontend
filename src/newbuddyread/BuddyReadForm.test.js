import { render } from "@testing-library/react";
import BuddyReadForm from "./BuddyReadForm";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <BuddyReadForm 
            formData={{
                buddy: "buddy@email.com"
            }}
        />
      </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
