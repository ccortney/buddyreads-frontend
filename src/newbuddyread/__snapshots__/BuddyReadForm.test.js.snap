// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    align="center"
    class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
  >
    <form>
      <div
        class="MuiBox-root css-1l4w6pd"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-ekfslj-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="buddy"
            id="buddy-label"
          >
            Buddy's Email
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="buddy"
              name="buddy"
              type="text"
              value="buddy@email.com"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1in441m"
              >
                <span>
                  Buddy's Email
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiBox-root css-kgh0nc"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Invite
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
